src/main/java/io/warp10/script/ext/influxdb/HttpClientUtils.java

```diff
@@ -0,0 +1,39 @@
+package io.warp10.script.ext.influxdb;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import io.warp10.script.WarpScriptException;
+import okhttp3.OkHttpClient.Builder;
+
+public class HttpClientUtils {
+
+    private static final String KEY_READ_TIMEOUT = "readTimeout";
+    private static final String KEY_WRITE_TIMEOUT = "writeTimeout";
+
+    public static Builder getOkHttpClientBuilder(final String functionName, final Map<Object, Object> params) throws WarpScriptException {
+
+        final Builder result = new Builder();
+
+        final Long readTimeout = getLongValueFromKey(functionName, params, KEY_READ_TIMEOUT);
+        if (readTimeout != null) {
+            result.readTimeout(readTimeout, TimeUnit.MILLISECONDS);
+        }
+
+        final Long writeTimeout = getLongValueFromKey(functionName, params, KEY_WRITE_TIMEOUT);
+        if (writeTimeout != null) {
+            result.writeTimeout(writeTimeout, TimeUnit.MILLISECONDS);
+        }
+
+        return result;
+    }
+
+    private static Long getLongValueFromKey(final String functionName, final Map<Object, Object> params, final String key) throws WarpScriptException{
+        try {
```

Comment:
Use a conditional testing if params.get(key) is an instance of of Long rather than relying on catching a ClassCastException, and pass a default value in getLongValueFromKey (from the Builder instance), so the code is easier to understand.

Reply:
I take into account your observations. 
For the second part, I'm not sure I coded what you explained to me, but I tried. :)